src/Window.cpp

```diff
@@ -237,13 +243,28 @@ bool Gosu::Window::tick()
     }
     
     SDL_Event e;
+
     while (SDL_PollEvent(&e)) {
-        if (e.type == SDL_QUIT) {
-            SDL_HideWindow(shared_window());
-            return false;
-        }
-        else {
-            input().feed_sdl_event(&e);
+        switch (e.type) {
+            case (SDL_QUIT): {
+                SDL_HideWindow(shared_window());
+                return false;
+            }
+        #if SDL_VERSION_ATLEAST(2, 0, 5)
+            case (SDL_DROPBEGIN):
+            case (SDL_DROPCOMPLETE):
+            case (SDL_DROPTEXT):
+        #endif
+            case (SDL_DROPFILE): {
+                char* dropped_filedir = e.drop.file;
```

Comment:
If you change this to `std::string`, please add a null check - sadly, `std::string(nullptr)` crashes instead of constructing an empty string.